frontend/src/views/CellCards/components/CellCard/components/EnrichedGenesTable/index.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useEffect, useMemo, useState } from "react";
+import {
+  TableTitle,
+  TableTitleWrapper,
+  TableUnavailableContainer,
+  TableUnavailableHeader,
+  TableUnavailableDescription,
+  TableTitleInnerWrapper,
+} from "../common/style";
+import Link from "../common/Link";
+import { TableTitleOuterWrapper } from "./style";
+import Table from "../common/Table";
+import DropdownSelect from "../common/DropdownSelect";
+import { SelectChangeEvent } from "@mui/material/Select";
+import { useEnrichedGenes } from "src/common/queries/cellCards";
+import HelpTooltip from "../common/HelpTooltip";
+import { ROUTES } from "src/common/constants/routes";
+
+export const CELL_CARD_ENRICHED_GENES_TABLE = "cell-card-enriched-genes-table";
+export const CELL_CARD_ENRICHED_GENES_TABLE_DROPDOWN =
+  "cell-card-enriched-genes-table-dropdown";
+
+interface TableRow {
+  symbol: string;
+  name: string;
+  me: string;
+  pc: string;
+}
+const tableColumns: Array<keyof TableRow> = ["symbol", "name", "me", "pc"];
+
+const tableColumnNames: Record<keyof TableRow, string> = {
+  symbol: "Symbol",
+  name: "Name",
+  me: "Expression Score",
+  pc: "% of Cells",
+};
+
+interface Props {
+  cellTypeId: string;
+}
+
+const EnrichedGenesTable = ({ cellTypeId }: Props) => {
+  const { data: genes } = useEnrichedGenes(cellTypeId);
+  const [selectedOrganism, setSelectedOrganism] = useState("");
+
+  const uniqueOrganisms = useMemo(() => {
+    if (!genes) return [];
+    const organisms = new Set<string>();
+    for (const markerGene of genes) {
+      organisms.add(markerGene.organism);
+    }
+    if (!selectedOrganism && Array.from(organisms).includes("Homo sapiens"))
+      setSelectedOrganism("Homo sapiens");
+    else if (!selectedOrganism)
+      setSelectedOrganism(Array.from(organisms).at(0) ?? "");
+
+    const organismsArray = Array.from(organisms).sort((a, b) => {
+      if (a === "Homo sapiens") return -1;
+      if (b === "Homo sapiens") return 1;
+      return a.localeCompare(b);
+    });
+    return organismsArray;
+  }, [genes, cellTypeId]);
+
+  const tableRows: TableRow[] = useMemo(() => {
+    if (!genes) return [];
+    const rows = [];
+    for (const markerGene of genes) {
+      const { pc, me, name, symbol, organism } = markerGene;
+      if (organism !== selectedOrganism) continue;
+      rows.push({
+        symbol: symbol,
+        name,
+        me: me.toFixed(2),
+        pc: (pc * 100).toFixed(1),
+      });
+    }
+    return rows;
+  }, [genes, selectedOrganism, cellTypeId]);
+
+  useEffect(() => {
+    return () => {
+      setSelectedOrganism("");
+    };
+  }, []);
```

Comment:
This is a cleanup function, right? Since this is a bit fringe, might want to comment the what and why